### ISSUE TITLE
Native SSO Issue when deeplink launching Uber crashes via deeplink uri

### ISSUE THREAD

**Issue by theonetheycallneo** [uber/rides-android-sdk#126]
Github issues are for bug reports. If this is a question around usage or understanding please use
Stack Overflow. https://stackoverflow.com/questions/tagged/uber-api

Library version:  com.uber.sdk:rides-android:0.9.1

Repro steps, stacktrace, screenshots:

We have implemented the Single Sign On for Android.

However, when we invoke the deeplink, it launches Uber app, then Uber app hangs/crashes and sends us back in ours.

https://youtu.be/nAMfsW8ByLk

Here's the generated sso url from the sdk:

```
uber://connect?client_id=tpU1890vz3bf-QiY1H-Fs5AKBfZ5C62E&scope=all_trips history places profile request request_receipt&sdk=android&sdk_version=3.0.0
```

Any ideas?

Expected Behavior:  

We expect the Single Sign On to auth the user then loop back and launch our app.

**Comment by saurabhdhillon** [comment]
I am also experiencing the same problem with the difference that uber hangs to a white screen and returns to calling activity on back presses.

I have tried:
Method 1:
```
SessionConfiguration config = new SessionConfiguration.Builder()
                .setClientId("VLEPmJ4-yqMuwFGTFWCR_9rpzFFpFi1-")
                .setClientSecret("GlA71zoYErPJoiuPWsEoMC7UegEQcdCgRIwd-l9J")
                .setServerToken("8mwj6-XEzqZCVer93LNe0QgoFiP8PejNrFSD87Kz")
                .build();

        Location location = Singleton.getInstance(context).getOneTimeLocation().getLocation();
        RideParameters rideParams = new RideParameters.Builder()
                .setPickupLocation(location.getLatitude(), location.getLongitude(), "Your", "Your")
                .setDropoffLocation(latitude, longitude, address, "")
                .build();

        RideRequestDeeplink deeplink = new RideRequestDeeplink.Builder(context)
                .setSessionConfiguration(config)
                .setFallback(Deeplink.Fallback.MOBILE_WEB)
                .setRideParameters(rideParams)
                .build();

       // deeplink.execute();
        
        try {
            Intent intent = new Intent(Intent.ACTION_VIEW);
            intent.setData(deeplink.getUri());
            ((BaseActivity) context).startActivity(intent, null, false);
        } catch (Exception e) {

        }
```

Method 2:
`String baseUrl = "https://m.uber.com/ul/?action=setPickup&";
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append(baseUrl);
        stringBuilder.append("client_id=").append("VLEPmJ4-yqMuwFGTFWCR_9rpzFFpFi1-").append("&");
        stringBuilder.append("pickup[latitude]=").append(location.getLatitude()).append("&");
        stringBuilder.append("pickup[longitude]=").append(location.getLongitude()).append("&");
        stringBuilder.append("dropoff[latitude]=").append(latitude).append("&");
        stringBuilder.append("dropoff[longitude]=").append(longitude).append("&");
        stringBuilder.append("dropoff[formatted_address]=").append(address);
        //Opened this url in browser ending in same result
`

Both the methods had same result, Uber app hangs with white screen. 

**Note: In Method 2, it provides me with _Open link with chrome or uber_, if i choose _Just Once_ it opens correctly but if it choose _Always_ option it keeps redirecting me to hang Uber.**

**Comment by george-freebirdrides** [comment]
Also experiencing this behavior. +1

**Comment by tyvsmith** [comment]
What is the logcat behavior during this? Does it occur every time? Is it for all deeplinks into the uber app?

**Comment by theonetheycallneo** [comment]
@tyvsmith This is during the single sign on authentication or a deeplink to launch the uber app with a ride.

Unfortunately, since we cannot login, the user is unable to logout.

**Comment by tyvsmith** [comment]
@theonetheycallneo Can you demonstrate your config and calling code? It seems odd that both are crashing for you, so I'm wondering about Activity Flags or any custom logic. Does logcat show anything? The example code shown doesn't indicate that it's URI encoded, so another consideration is that it's crashing on parsing that. Sorry for the delays here, this is understaffed at the moment, but I'll try to dig in if I can get some more data.

**Comment by theonetheycallneo** [comment]
@here I'm closing this issue.

We got confused using our two environments dev and production.  We needed to create 2 keys for each and copy them both to the uber dashboard.

For those of you struggling, our dev key uses the base Android Studio one:

```
cat ./android/keystores/debug.keystore.properties
key.store=debug.keystore
key.alias=androiddebugkey
key.store.password=android
key.alias.password=android
```

alias: androiddebugkey
password: android

Then copy that into your dashboard to test on device locally.
